lib/exec.js

```diff
@@ -14,8 +14,9 @@ module.exports = function exec(script, program) {
   if (firstArg !== undefined) {
     argvIndex = process.argv.indexOf(firstArg);
   }
-  var argv = argvIndex === -1 ? [] : process.argv.splice(argvIndex + 1),
-      command = (typeof script === 'string' ? script : script.command) + ' ' + argv.join(' ');
+  var argv = argvIndex === -1 ? [] : process.argv.splice(argvIndex + 1)
+    .map(function(arg) { return arg.indexOf(' ') >= 0 ? '"'+ arg + '"' : arg });
+  var command = (typeof script === 'string' ? script : script.command) + ' ' + argv.join(' ');
```

Comment:
I would prefer if this is handled using if statemens, it's hard to read this way